modules/ROOT/pages/instanton-limitations.adoc

```diff
@@ -161,6 +162,72 @@ This configuration uses placeholder values for things like the database name, ho
+[source,java]
+----
+  @Inject
+  @ConfigProperty(name = "static_config")
+  String staticConfig;
+
+  @Inject
+  @ConfigProperty(name = "static_optional_config")
+  Optional<String> staticOptionalConfig;
+
+  @Inject
+  @ConfigProperty(name = "dynamic_config")
+  Provider<String> dynamicConfig;
+
+  @Inject
+  @ConfigProperty(name = "dynamic_optional_config")
+  Provider<Optional<String>> dynamicOptionalConfig;
+----
+The injected `static_config` will cause an error like the following example if there is no value found in an existing config source during checkpoint.
+
+[source,sh]
+----
+SRCFG02000: Failed to Inject @ConfigProperty for key static_config into io.example.Example.staticConfig since the config property could not be found in any config source.
+----
+
+This error can be avoided by providing a default value for the configuration key in one of the following ways:
+
+Specify the default value on the @ConfigProperty annotation like the following:
+[source,java]
+----
+  @Inject
+  @ConfigProperty(name = "static_config", defaultValue = "defaultValue")
+  String staticConfig;
+----
+
+Specify the default value in the application `META-INF/microprofile-config.properties` resource.
+[source,sh]
+----
+  static_config=defaultValue
+----
+
+Specify a default value in a server.xml variable.
+[source,xml]
+----
+  <variable name="static_config" defaultValue="defaultValue" />
+----
+
+The `defaultValue` can then be overridden on restore by configuring the value with an environment variable or values from the variable source directory.
+
+However, if no default value is set, we can still avoid the above error by injecting config using `static_optional_config`, `dynamic_config`, `dynamic_optional_config`. But caution is needed for CDI beans that are `@ApplicationScoped` which are created during application startup when using the afterAppStart checkpoint option. Trying to access the `dynamic_config` early during the application startup at checkpoint can cause the following error:
+[source,sh]
+----
+java.util.NoSuchElementException: SRCFG00014: The config property dynamic_config is required but it could not be found in any config source.
+----
+While accessing the `static_optional_config` and `dynamic_optional_config` can cause the following error:
+[source,sh]
+----
+java.util.NoSuchElementException: No value present
+----
+
+Therefore, it is advisable to have a default value set for injected config properties in order to avoid the above mentioned errors. Furthermore, if the @ConfigProperty injection site is not using dynamic config then any default value injected into the application scoped bean before checkpoint will not be updated on restore. This is already mentioned in the docs at xref:#mp-config[Accessing MicroProfile Config too early]
```

Comment:
```suggestion
To avoid these errors, set a default value for injected config properties. Furthermore, if the `@ConfigProperty` injection site is not using dynamic configuration, then any default value that is injected into the application-scoped bean before checkpoint is not updated on restore. For more information, see xref:#mp-config[Accessing MicroProfile Config too early]
```